### DISCUSSION TITLE
[C++] Can I write a C++ rule?

### DISCUSSION THREAD

**Discussion by cyw3** [pmd/pmd#3034] [Q&A] [answered]
I can not find C++ rules.
and when I use xpath to custom rules, I find: 
![image](https://user-images.githubusercontent.com/11549103/48751782-1df15a80-ecc1-11e8-9f7f-684137f771d3.png)

**Comment by jsotuyod** [comment, marked as answer]
@cyw3 for the time being C/C++ is a CPD-only language.

Adding full support is possible, and shouldn't be too hard as we already have a JavaCC grammar, but needs to be done. If you are willing to tackle that, feel free to contact us on gitter https://gitter.im/pmd/pmd I'll gladly give you some pointers.
